libnavui-maps/src/test/java/com/mapbox/navigation/ui/maps/internal/route/line/MapboxRouteLineUtilsTest.kt

```diff
@@ -872,101 +831,61 @@ class MapboxRouteLineUtilsTest {
 
     @Test
     fun getRouteLineTrafficExpressionDataMissingRoadClass() {
-        val routeAsJsonJson = FileUtils.loadJsonFixture(
+        val route = loadNavigationRoute(
             "route-with-missing-road-classes.json"
         )
-        val route = DirectionsRoute.fromJson(routeAsJsonJson)
 
         val result = MapboxRouteLineUtils.extractRouteDataWithTrafficAndRoadClassDeDuped(
             route,
             MapboxRouteLineUtils.getRouteLegTrafficCongestionProvider
         )
 
-        assertEquals(7, result.size)
-        assertEquals(0.0, result[0].distanceFromOrigin, 0.0)
+        assertEquals(8, result.size)
+        assertEquals(0.0, result[0].offset, 0.0)
         assertEquals("severe", result[0].trafficCongestionIdentifier)
         assertEquals("motorway", result[0].roadClass)
-        assertEquals(3.7, result[1].distanceFromOrigin, 0.0)
+        assertEquals(0.00236427570458575, result[1].offset, 0.0)
         assertEquals("unknown", result[1].trafficCongestionIdentifier)
         assertEquals("motorway", result[1].roadClass)
-        assertEquals(27.5, result[2].distanceFromOrigin, 0.0)
+        assertEquals(0.01736459676474489, result[2].offset, 0.0)
         assertEquals("severe", result[2].trafficCongestionIdentifier)
         assertEquals("motorway", result[2].roadClass)
-        assertEquals(271.8, result[3].distanceFromOrigin, 0.0)
+        assertEquals(0.1714843563045706, result[3].offset, 0.0)
         assertEquals("severe", result[3].trafficCongestionIdentifier)
         assertEquals("intersection_without_class_fallback", result[3].roadClass)
-        assertEquals(305.2, result[4].distanceFromOrigin, 0.0)
+        assertEquals(0.19250306326360345, result[4].offset, 0.0)
         assertEquals("severe", result[4].trafficCongestionIdentifier)
         assertEquals("motorway", result[4].roadClass)
-        assertEquals(545.6, result[5].distanceFromOrigin, 0.0)
+        assertEquals(0.34420704878106756, result[5].offset, 0.0)
         assertEquals("severe", result[5].trafficCongestionIdentifier)
         assertEquals("intersection_without_class_fallback", result[5].roadClass)
-        assertEquals(1168.3000000000002, result[6].distanceFromOrigin, 0.0)
+        assertEquals(0.7375706868313839, result[6].offset, 0.0)
         assertEquals("severe", result[6].trafficCongestionIdentifier)
         assertEquals("motorway", result[6].roadClass)
-    }
-
-    @Test
-    fun getRouteLineExpressionDataWithStreetClassOverrideWhenHasStreetClassesOnMotorwayMultiLeg() {
```

Comment:
Duplicate offsets are now allowed to appear in the result of `extractRouteData`, they are filtered out on expression creation level. I moved to a different class to cover this specific case.